components/datetime/src/format/datetime.rs

```diff
@@ -221,23 +230,62 @@ where
             field.length,
         )?,
         FieldSymbol::DayPeriod(period) => {
-            let symbol = symbols.get_symbol_for_day_period(
-                period,
-                field.length,
-                datetime.datetime().hour().ok_or(Error::MissingInputField)?,
-                arithmetic::is_top_of_hour(
-                    pattern,
-                    datetime.datetime().minute().map(u8::from).unwrap_or(0),
-                    datetime.datetime().second().map(u8::from).unwrap_or(0),
-                ),
-            );
+            let symbol = symbols
+                .expect("Expect symbols to be present")
+                .get_symbol_for_day_period(
+                    period,
+                    field.length,
+                    datetime.datetime().hour().ok_or(Error::MissingInputField)?,
+                    arithmetic::is_top_of_hour(
+                        pattern,
+                        datetime.datetime().minute().map(u8::from).unwrap_or(0),
+                        datetime.datetime().second().map(u8::from).unwrap_or(0),
+                    ),
+                );
             w.write_str(symbol)?
         }
         field @ FieldSymbol::TimeZone(_) => return Err(Error::UnsupportedField(field)),
     };
     Ok(())
 }
 
+// This function determins whether the struct will load symbols data.
+// Keep it in sync with the `write_field` use of symbols.
```

Comment:
Issue: "Keep it in sync" is a noble comment, but the reality is that devs don't obey instructions in code comments unless there is a hard error.  Is there a way to make this into more of a hard error?  I guess test coverage, to make sure we hit your `.expect()`, but is that all we can do?

Reply:
I assume it is. I could try to build some more elaborate enum that has to be true here and in the analyze function together, but I think it's overengineering at this point, and I'd prefer to land it as is.